rasa/nlu/tokenizers/tokenizer.py

```diff
@@ -0,0 +1,39 @@
+from typing import Text, List, Optional, Dict, Any
+
+from rasa.nlu.constants import (
+    MESSAGE_RESPONSE_ATTRIBUTE,
+    MESSAGE_TEXT_ATTRIBUTE,
+    CLS_TOKEN,
+)
+
+
+class Token(object):
+    def __init__(self, text, offset, data=None):
+        self.offset = offset
+        self.text = text
+        self.end = offset + len(text)
+        self.data = data if data else {}
+
+    def set(self, prop, info):
+        self.data[prop] = info
+
+    def get(self, prop, default=None):
+        return self.data.get(prop, default)
+
+
+class Tokenizer(object):
+    def add_cls_token(
+        self,
+        tokens: List[Token],
+        use_cls_token: bool,
```

Comment:
why don't we make it a property of `Tokenizer`?

Reply:
it's a bit complicated as all tokenizers inherit from `Tokenizer` and `Component`. And we still want to be able to set the value via the component config. Will try to figure something out.

Reply:
found a way

Reply:
here: https://github.com/RasaHQ/rasa/blob/2951bd8acfb9f2915ac97f78e4e585944837a461/rasa/core/featurizers.py#L269, I also use multi inheritance and both init are called

Reply:
should they inherit only from `Tokenizer` and `Tokenizer` inherit from `Component`?

Reply:
yes, the order is important, now it should work

Reply:
yes, let's do that, it would also be more consistent with the other components